lib/Service/MetadataService.php

```diff
@@ -181,6 +182,31 @@ protected function readExif($file) {
         return @exif_read_data($file, 0, true);
     }
 
+    protected function readPngChunks($file) {
+        $return = array();
+        $fp = fopen($file, 'r');
+        if (fread($fp, 8) !== "\x89\x50\x4e\x47\x0d\x0a\x1a\x0a") {
+            throw new \Exception($this->t('This image is not a png'));
+        }
+        while ($chunkHeader = fread($fp, 8)) {
+            $chunk = unpack('Nsize/a4type', $chunkHeader);
+            $data = (0 < $chunk['size']) ? fread($fp, $chunk['size']) : '';
```

Comment:
Do we really need to read the data in all cases (regardless of chunk type)?

Reply:
Fixed this issue.